<launch>

  <!-- Args -->
  <arg name="namespace" default=""/>
  <arg name="param_file" default="$(find-pkg-share sensors)/config/drivers.yaml"/>
  <arg name="log_level"  default="info"/>

  <!-- Enable/disable individual drivers -->
  <arg name="bno055_enable" default="true"/>

  <!-- BNO055 IMU -->
  <node pkg="sensors" exec="bno055_node" name="bno055" namespace="$(var namespace)"
        output="screen" if="$(var bno055_enable)"
        args="--ros-args --log-level $(var log_level)">
    <param from="$(var param_file)"/>
    <!-- Example remaps (uncomment if you need) -->
    <!-- <remap from="imu/data" to="imu/raw"/> -->
  </node>

  <!-- Add more real hardware drivers here (e.g., lidar) -->
</launch>
